javalin/src/main/java/io/javalin/http/staticfiles/JettyResourceHandler.kt

```diff
@@ -52,6 +59,49 @@ class JettyResourceHandler(val precompressStaticFiles: Boolean = false) : io.jav
         }
     }
 
+    inner class AliasResourceHandler(private val urlPathPrefix: String, private val aliasCheck: AliasCheck) : ResourceHandler() {
+        override fun getResource(path: String): Resource {
+            val targetResource by lazy { path.removePrefix(urlPathPrefix) }
+            return when {
+                urlPathPrefix == "/" -> getResourceWithAliasChecks(path)!!
+                targetResource == "" -> getResourceWithAliasChecks("/")!! // directory without trailing '/'
+                !path.startsWith(urlPathPrefix) -> EmptyResource.INSTANCE
+                !targetResource.startsWith("/") -> EmptyResource.INSTANCE
+                else -> getResourceWithAliasChecks(targetResource)!!
+            }
+        }
+
+        // same as regular ResourceHandler but with alias checks
+        private fun getResourceWithAliasChecks(path: String) : Resource? {
+            try {
+                var resource: Resource? = null
+                if (baseResource != null) {
+                    val caconicalPath = URIUtil.canonicalPath(path)
+                    resource = baseResource.addPath(caconicalPath)
+                    if (resource != null && resource.isAlias && !checkAlias(path, resource)) {
+                        return null
+                    }
+                }
+                if ((resource == null || !resource.exists()) && path.endsWith("/jetty-dir.css")) resource = stylesheet
```

Comment:
What is this?

Reply:
Already answered above:
I just code copied original jetty's implementation of ResourceHanlder.getResource() and changed only alias checks. I decided to keep the part with /jetty-dir.css (to make sure the new functionallity won't affect the old)

Reply:
> Already answered above:

Sorry, I missed that. 

> Should I remove it?

Is it possible to only use the aliashandler if there's an alias, and call `super` otherwise?

Reply:
I don't think so, because to know if there is an alias we should call 50% of lines from `super` and then if no alias call again `super`? Looks like overhead?
```
if (baseResource != null) {
    val caconicalPath = URIUtil.canonicalPath(path)
    resource = baseResource.addPath(caconicalPath)
    if (!resource.isAlias) {
        return super.getResource() // here we again get caconicalPath and call addPath
    }
}
```

Reply:
Okay. Let's keep it like this for now then.